drv/lpc55-update-server/src/main.rs

```diff
@@ -169,6 +177,19 @@ impl idl::InOrderUpdateImpl for ServerImpl {
             version: HUBRIS_BUILD_VERSION,
         })
     }
+
+    fn status(
+        &mut self,
+        _: &RecvMessage,
+    ) -> Result<UpdateStatus, RequestError<Infallible>> {
+        // Safety: Data is published by stage0
+        let addr: &[u8] = unsafe { BOOTSTATE.assume_init_ref() };
```

Comment:
I mentioned this before but I had bad luck with the compiler still trying to optimize calls to `assume_init` out. Maybe it works better with `assume_init_ref` because it's shared?

Reply:
I will talk to Cliff about it next week. I don't plan to merge before then anyway.

Reply:
I'd be happy to walk you through how to check for this condition.